modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java

```diff
@@ -38,30 +39,30 @@
 /**
  * @author Yasuyuki Takeo
  */
-@Component(immediate = true, service = DeepLClient.class)
-public class DeepLClient {
+@Component(immediate = true, service = TranslationManager.class)
+public class TranslationManager {
 
-	public TranslateResponse execute(
+	public List<Translation> getTranslateResponse(
 			String authKey, String text, String sourceLanguageId,
 			String targetLanguageId, String url)
 		throws IOException, PortalException {
 
-		return JSONUtil.toObject(
-			_fetch(authKey, text, sourceLanguageId, targetLanguageId, url),
-			TranslateResponse.class);
-	}
-
-	public List<SupportedLanguage> getSupportedLanguages(
-			String authKey, String target, String url)
-		throws IOException, PortalException {
-
-		return JSONUtil.toObject(
-			_getSupportedLanguage(authKey, target, url),
-			new TypeReference<List<SupportedLanguage>>() {
-			});
+		JSONObject jsonObject = _jsonFactory.createJSONObject(
+			post(authKey, text, sourceLanguageId, targetLanguageId, url));
+
+		return JSONUtil.toList(
+			jsonObject.getJSONArray("translations"),
+			customFieldJSONObject -> new Translation() {
+				{
+					detectedSourceLanguage = customFieldJSONObject.getString(
+						"detected_source_language");
+					text = customFieldJSONObject.getString("text");
+				}
+			},
+			_log);
 	}
 
-	private String _fetch(
+	protected String post(
```

Comment:
Don't use protected methods unless you need them for subclasses or unit tests. Always prefer private methods/fields.